### PR TITLE
[tensor-widget] Add colormap selection and Ctrl/Alt/Shift+wheel zooming

### PR DESCRIPTION
**Pull request by caisq**
* Motivation for features / changes
  * Continue developing TensorWidget

* Technical description of changes
  * Support Ctrl/Alt/Shift + mouse wheel zooming
  * Add an optional field to TensorWidgetOptions: wheelZoomKey to allow specifying whether alt, ctrl or shift key activates wheel zooming.
  * Add jet colormap
  * Add colormap selection menu
  * Added a color bar in the value tooltip that indicates where the currently pointed-to value is within the color scale.

* Screenshots of UI changes
  * image
  * image
  * image
  * image

### CODE REVIEW

**Comment by wchargin** [comment]
In the rest of TensorBoard we use Alt+Wheel for zooming to avoid
conflicting with standard browser interfaces. Maybe consider following
that precedent here?

**Comment by caisq** [comment]
@wchargin Thanks for the info. I added an optional field to TensorWidgetOptions: wheelZoomKey to allow specifying whether alt, ctrl or shift key activates wheel zooming.

**Comment by caisq** [comment]
Gentle ping @bileschi @wchargin

File: tensorboard/components/tensor_widget/colormap.ts [hunk 1]
```diff
@@ -17,6 +17,21 @@ limitations under the License.
...
+export interface ColorMapConfig {
+  /**
+   * Minimum value that the color map can map to without clipping.
+   * Must be a finite value.
+   */
+  min: number;
+
+  /**
+   * Minimum value that the color map can map to without clipping.
+   * Must be a finite value and be >= `min`.
```
Comment (bileschi) [inline tensorboard/components/tensor_widget/colormap.ts]: What happens in the case of equality?

**Comment by bileschi** [comment]
The native github review tool is very bad.  Let's re-open the suggestion to use reviewable.

**Comment by wchargin** [comment]
I won’t have bandwidth to get to this shortly. Let me know if you need
anything specific and bounded.

File: tensorboard/components/tensor_widget/colormap.ts (hunk 1 above)
Comment (caisq) [reply tensorboard/components/tensor_widget/colormap.ts]: Then all finite values will mapped to a relative value of 0.5 and in turn be mapped to the mid-point of the colormap. Added comment here.

**Comment by caisq** [comment]
Thanks, @bileschi and @wchargin

**Comment by caisq** [comment]
cc @mndevec

**Pull request merged**